docs/src/reference/asciidoc/core/kerberos.adoc

```diff
@@ -0,0 +1,481 @@
+{krb} principal to one of these realms, set the new user's username to just the main part of the principal name, without
+any realm or host information. For instance, `client@REALM` would just be `client` and `someservice/domain.name@REALM`
+would just be `someservice`.
+
+Once you have configured your client principals so that they can be impersonated in {es}, you must create a role that
+Hive will use to impersonate those users.
+
+[source,js]
+----
+PUT _xpack/security/role/hive_proxier
+{
+  "run_as": ["*"] <1>
+}
+----
+// CONSOLE
+<1> Using `"*"` is very permissive, so it is best to limit this to only the users which Hive should impersonate
+
+Now that there are users to impersonate, and a role that can impersonate them, make sure to map the Hive principal to
+the proxier role, as well as any of the roles that the users it is impersonating would have. This allows the Hive
+principal to create and read indices, documents, or do anything else its impersonated users might be able to do.
+While hive is impersonating the user, it must have these roles or else it will not be able to fully impersonate that
+user.
+
+[source,js]
+----
+POST /_xpack/security/role_mapping/hive_hiveserver2_mapping
+{
+  "roles": [
+    "client_role_1", <1>
+    "client_role_2",
+    "hive_proxier" <2>
+  ],
+  "enabled": true,
+  "rules": {
+    "field" : { "username" : "hive/hiveserver2.address@REALM" } <3>
+  }
+}
+----
+// CONSOLE
+<1> Any number of roles that the clients that Hive must impersonate would have
+<2> The added role that allows Hive to impersonate users
+<3> The name of the hive server principal to match against
+
+Additionally, if managing these role mappings via the API's is not desired, they can instead be managed in a
+https://www.elastic.co/guide/en/elastic-stack-overview/current/mapping-roles.html#mapping-roles-file[role mapping file].
+
+[[kerberos-hive-running]]
+==== Running your Hive queries
+
+Once all user accounts are configured and all previous steps for enabling {krb} auth in Hadoop and Hive are complete,
+there should be no differences in creating Hive queries from before.
+
+[[kerberos-pig]]
+=== {krb} on Pig
+
+[[kerberos-pig-requirements]]
+==== Requirements
+
+Before using {krb} authentication to {es} in Pig, Hadoop Security must be enabled. Make sure you have done all the
+required steps for <<kerberos-hadoop-requirements,configuring your Hadoop cluster>> for use with {es} and {krb}.
+
+If {eh} is configured for {krb} authentication and Hadoop security is enabled, {eh}'s storage functions in Pig will
+automatically obtain delegation tokens for jobs when submitting them to the cluster.
+
+[[kerberos-spark]]
+=== {krb} on Spark
+
+[[kerberos-spark-requirements]]
+==== Requirements
+
+Using {krb} authentication in {eh} for Spark has the following requirements:
+
+1. Your Spark jobs must be deployed on YARN. Using {krb} authentication in {eh} does not support any other Spark cluster
```

Comment:
Interesting, so this won't work with any Spark standalone cluster?

Reply:
Unfortunately it will not - All of Spark's credential features rely on Hadoop mechanisms for transmission. My understanding at the moment is that as of 2.3.0, they moved the delegation token management classes for HDFS, HBase and Hive to the core Spark packages. Those management classes are not open for extension by third-party dependencies. The service credential provider interfaces that we are able to extend are only available in the YARN deployment.